syntax="proto3";
package streamDL;


// streamDLbroker
service streamDLbroker{
    // deploy model
    rpc set_deploy_model(stream Model) returns (Reply) {}

    // return deployed model list
    rpc get_deployed_model(null) returns (ModelList) {}

    // check duplicate check
    rpc is_deployed_model(ModelName) returns (Reply) {}

    // return ami lists available in the system
    rpc get_ami_list(null) returns (AMIList) {}
}

message Chunk {
    bytes buffer = 1;
}

message ModelName{
    string name = 1;
}

message ModelList{
    repeated Model model = 1;
}

message Model{
    string name = 1;
    repeated string stream_inputs = 2;
    bytes buffer = 3;
}

message ModelInfo{
    string name = 1;
    repeated string stream_inputs = 2;
    InputFmt input_fmt = 3;
}

message AMIList{
    repeated string ami_id = 1;
}

message InputFmt{
    int32 input_window_size = 1;
    int32 output_shift = 2;
    int32 output_window_size = 3;
}

message Reply {
    bool status = 1;
    string message = 2;
}

message null{
}

